brave/browser-laptop
# Add sync

this adds basic support for browser sync, using https://github.com/brave/sync

Fix #1854

# Test Plan:

## setup
- make sure automated tests related to sync are passing
- `npm install` to add dependencies
- `npm run download-sync-client` to download the sync client to the right subdirectory
- add the following to package.json to create a separate browser 'device' to sync to:
```
 "start2": "node ./tools/start.js --user-data-dir=brave-development-2 --debug=5859 --enable-logging --v=0 --enable-extension-activity-logging --enable-sandbox-logging --enable-dcheck"
```

## test
- npm start
- go to about:preferences#sync
- enable sync
- restart, then click 'add device...'
- npm run start2
- in device 2, go to about:preferences#sync > i have an existing code > enter the code from device 1
- go to any website in either of the browsers, bookmark it and make some changes to the siteSettings. after a minute, it should show up in the other browser.

# todos before public release (not necessarily before merge)
- https://github.com/brave/sync/milestone/2
